packages/components/src/Banner/Banner.tsx

```diff
@@ -19,9 +19,11 @@ type Props = {
    */
   color?: NotificationVariant;
   /**
-   * The contents of the banner in addition to the icon
+   * The text content of the banner.
+   *
+   * Please note that this can contain text links, but block buttons should be passed in via the `action` prop.
    */
-  children: React.ReactNode;
+  textContent: React.ReactNode;
```

Comment:
I like your initial API in 448801c5897dbbc1177a8e0289c7253f8ad728a5

```jsx
<Banner
  {...restArgs}
  title={<Banner.Title as="h1">{heading}</Banner.Title>}
  message={<Banner.Body>{content}</Banner.Body>}
/>
```

although maybe we don't need to manually provide Banner.Body

```jsx
<Banner
  {...restArgs}
  title={<Banner.Title as="h1">{heading}</Banner.Title>}
  message={content}
/>
```

The PR description mentions why you changed it and went with this API

> But in that version, title and message don't really need to be passed in as separate props because we're not doing anything special to either of them.

One thing that is nice about keeping the separate `title` and `message` props (even if we're not doing anything special with either) is the logical separation. It may be more obvious what kind of content does and does not go in there.

Once we have a single prop

```jsx
textContent={
  <>
    <Banner.Title as="h1">{heading}</Banner.Title>{" "}
    <Banner.Body>{content}</Banner.Body>
  </>
}
```

It might be a little more natural to start putting other stuff (like buttons) in there too

```jsx
textContent={
  <>
    <Banner.Title as="h1">{heading}</Banner.Title>{" "}
    <Banner.Body>{content}</Banner.Body>
    <Button>click me!</Button>
  </>
}
```

But it's a small thing, so 🤷

Reply:
I agree ... I wonder if we could stick with the `title` and `message` props and get rid of `Banner.Title` and `Banner.Body`? Similar to what Andrew suggested:
```
<Banner {...restArgs} title={heading} message={content} />
```

Then internally we still render
```
// introduce a new headerAs prop as well
<Heading as={props.headerAs || "h2"} color={"inherit"} size="h3">
  {props.title}
</Heading>

<Text color={"inherit"} size="body">
  {props.message}
</Text>
```

Thoughts?

Reply:
Agreed that it would be nice to get rid of the Banner.Title and Banner.Body components (could still have internal components that wrap what people pass in, as Annie suggested).

Not sure if there are any ramifications of that or not, though.

Reply:
I think `message` would need to accept a ReactNode type anyways because of the link (and maybe both should allow it to be more flexible), so that case would still be fine. but I see how `textContent` makes more sense there

Reply:
I'm also fine with the idea of using `title` and `message` even if they don't make any styling changes to the content just to make it clearer what we expect you to pass in.

Reply:
I guess I'm just talking about the first commit https://github.com/chanzuckerberg/edu-design-system/commit/448801c5897dbbc1177a8e0289c7253f8ad728a5, but with renaming `Banner.Body` per @anniehu4 's other comment https://github.com/chanzuckerberg/edu-design-system/pull/622#discussion_r706338844

Reply:
I have mild concern that people will be confused by having to pass `title={<Banner.Title>...</Banner.Title>}` and it'll lead to more custom styling 🤔  so I'd prefer one `textContent` prop over that

Reply:
Ah, I see. Then let's go with `textContent` for now and see how it goes.